native/wasmex/src/component_instance.rs

```diff
@@ -150,13 +154,13 @@ fn term_to_val(param_term: &Term, param_type: &Type) -> Result<Val, Error> {
 
 fn term_to_field_name(key_term: &Term) -> String {
     match key_term.get_type() {
-        TermType::Atom => key_term.atom_to_string().unwrap(),
-        _ => key_term.decode::<String>().unwrap(),
+        TermType::Atom => key_term.atom_to_string().unwrap().to_case(Case::Kebab),
+        _ => key_term.decode::<String>().unwrap().to_case(Case::Kebab),
```

Comment:
looking at these `unwrap` calls: When/how would these trigger? can they be triggered through user input? Wouldn't be great if we crash the BEAM just because someone chose a weird field name 🤔